torchci/rockset/commons/__sql/flaky_workflows_jobs.sql

```diff
@@ -1,73 +1,114 @@
-WITH flaky_jobs AS (
+WITH dedups AS (
+  -- Note that there can be more than one commit with the same ID with the actual author and pytorchmergebot.
+  -- This mess up the results in some cases, so this removes all redundant information and only keeps what is
+  -- needed for the later query
   SELECT
+    DISTINCT CONCAT(w.name, ' / ', job.name) AS fullname,
     w.name AS workflow_name,
+    w.id AS workflow_id,
     job.name AS job_name,
+    job.id AS job_id,
+    job.conclusion AS conclusion,
+    push.head_commit.id AS head_commit,
+    push.head_commit.timestamp AS head_commit_timestamp,
+    job.run_attempt AS run_attempt,
+    ROW_NUMBER() OVER(
+      PARTITION BY w.id,
+      w.name,
+      job.name
+      ORDER BY
+        job.run_attempt DESC
+    ) AS row_num,
+  FROM
+    commons.workflow_run w
+    JOIN commons.workflow_job job ON w.id = job.run_id HINT(join_strategy = lookup)
+    JOIN push ON push.head_commit.id = w.head_commit.id
+  WHERE
+    (
+      job._event_time >= CURRENT_DATE() - HOURS(: numHours)
+      OR : numHours = 0
+    )
+    AND w.head_repository.full_name = : repo
+    AND ARRAY_CONTAINS(
+      SPLIT(: branches, ','),
+      w.head_branch
+    )
+    AND ARRAY_CONTAINS(
+      SPLIT(: workflowNames, ','),
+      w.name
+    )
+    AND job.name NOT LIKE '%mem_leak_check%'
+    AND job.name NOT LIKE '%rerun_disabled_tests%'
```

Comment:
The good news is that @DanilBaibak has already added that inclusion list into PyTorch https://github.com/pytorch/pytorch/blob/master/.github/pytorch-probot.yml#L16-L21.  That same list is used here as part of this query via the `workflowNames` parameter.  On the other hand, `pull` and `trunk` workflows from the list currently has 3 flavors.  Take `pull` as an example:

1. The `pull` workflow itself
2. The `pull` workflow running in `mem_leak_check` mode, run daily
3. The `pull` workflow running in `rerun_disabled_tests` mode, also run daily

So there is a need for an exclusion list too.  It would be nice to have this exclusion logic codified upstream in the same `pytorch-probot.yml` file, but I guess this would come later in separate PRs